presto-main/src/main/java/io/prestosql/type/UuidOperators.java

```diff
@@ -199,7 +199,7 @@ public static boolean isDistinctFrom(
             if (left.isNull(leftPosition)) {
                 return false;
             }
-            return left.compareTo(leftPosition, 0, UUID.getFixedSize(), right, rightPosition, 0, UUID.getFixedSize()) != 0;
+            return UUID.compareTo(left, leftPosition, right, rightPosition) != 0;
```

Comment:
I think you want `!UUID. equalTo(left, leftPosition, right, rightPosition)`

Reply:
Thanks @dain 
Code is much cleaner now